### ISSUE TITLE
Image layout tracking for depth+stencil images at vkCmdEndRenderPass

### ISSUE THREAD

**Issue by Qining** [KhronosGroup/Vulkan-LoaderAndValidationLayers#2506]
Looks like there is bug for the layout transition tracking at vkCmdEndRenderPass. The `finalLayout` is not included in the tracking, so that I keep getting error messages from validation layers like:
```
DS(ERROR / SPEC): object: 0x557ab9082cb0 type: 6 msgNum: 6 - Cannot submit cmd buffer using image (0xdc) [sub-resource: aspectMask 0x4 array layer 2, mip level 0], with layout VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL when first use is VK_IMAGE_LAYOUT_DEPTH_STENCIL_READ_ONLY_OPTIMAL.

```
I can get rid of the error by using pipeline barrier to switch the layout and change the `finalLayout` from `VK_IMAGE_LAYOUT_DEPTH_STENCIL_READ_ONLY_OPTIMAL` to `VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL`, (which is the same layout used in `VkAttachmentReference`) in the attachment description when creating the `VkRenderPass`.

Nothing else changed, so I think there is a problem with `vkCmdEndRenderPass + DepthStencil image layout transition`.

**Comment by chrisforbes** [comment]
@Qining we do account for the final layout -- this is the (admittedly slightly misnamed) `TransitionFinalSubpassLayouts` call in `core_validation::CmdEndRenderPass.`

Do you have a test case which demonstrates this?

**Comment by Qining** [comment]
Thanks for reply, let me see if I can make a test case for this.

**Comment by karl-lunarg** [comment]
Closed in KhronosGroup/Vulkan-ValidationLayers#70
